cabal-install/src/Distribution/Client/ProjectPlanning.hs

```diff
@@ -3058,18 +3150,25 @@ pruneInstallPlanPass2 pkgs =
                     keepNeeded (CD.ComponentTest  _) _ = TestStanzas  `optStanzaSetMember` stanzas
                     keepNeeded (CD.ComponentBench _) _ = BenchStanzas `optStanzaSetMember` stanzas
                     keepNeeded _                     _ = True
+
                 in ElabPackage $ pkg {
                   pkgStanzasEnabled = stanzas,
-                  pkgLibDependencies   = CD.filterDeps keepNeeded (pkgLibDependencies pkg),
+                  pkgLibDependencies   = CD.mapDeps (\_ -> map addInternal) $ CD.filterDeps keepNeeded (pkgLibDependencies pkg),
                   pkgExeDependencies   = CD.filterDeps keepNeeded (pkgExeDependencies pkg),
                   pkgExeDependencyPaths = CD.filterDeps keepNeeded (pkgExeDependencyPaths pkg)
                 }
-              r@(ElabComponent _) -> r
+              (ElabComponent comp) ->
+                ElabComponent $ comp { compLibDependencies = map addInternal (compLibDependencies comp) }
         }
       where
+        addInternal (cid, _) = (cid, (cid `Set.member` inMemoryTargets))
```

Comment:
Some comment please, we ignore the value here because we inserted some bogus value before?
Can we add an assertion that we *really* only have a bogus value here?